### PR TITLE
[ACM-4384] defer unsafe Close vulnerability

### PR DESCRIPTION
**Pull request by subbarao-meduri**
This PR addresses the following issues
1. [ACM-4384](https://issues.redhat.com/browse/ACM-4384) - defer unsafe close vulnerability
2. Adds a diagnostic message when MCO reconcile is skipped waiting for MCH CR to be ready (upgrade)
3. Miscellaneous warnings

CC: @bjoydeep @berenss - (2) should help diagnose MCO reconcile wasn't working as we saw in in ROSA HCP env

### CODE REVIEW

**Review by saswatamcode — APPROVED** [review]
Thanks! This fixes a lot of issues! :)

File: loaders/dashboards/pkg/util/grafana_util.go [hunk 1]
```diff
@@ -61,14 +61,16 @@ func SetRequest(method string, url string, body io.Reader, retry int) ([]byte, i
...
-			klog.Info("failed to parse response body ", "error ", err)
-		}
-		return respBody, resp.StatusCode
-	} else {
+	if resp == nil {
 		return nil, http.StatusNotFound
 	}
+
+	respBody, err := io.ReadAll(resp.Body)
+	if err != nil {
```
Comment (douglascamata) [inline loaders/dashboards/pkg/util/grafana_util.go:69]: I think we can improve this to defer the body closure and also avoid the linting for not handling the error:

```go
defer func() {
  err := resp.Body.Close()
  if err != nil {
    // log the error?
  }
}()
```

This applies to all the `req.Body.Close()` statements.

**Review by douglascamata — CHANGES_REQUESTED** [review]

File: loaders/dashboards/pkg/util/grafana_util.go (hunk 1 above)
Comment (subbarao-meduri) [reply loaders/dashboards/pkg/util/grafana_util.go:76]: using this pattern now.

**Review by douglascamata — APPROVED** [review]
/lgtm

**Pull request merged**